SwrveSDK/SwrveClient.brs

```diff
@@ -1,1112 +1,1022 @@
-                        session_token : ""
-                        isQAUser: false
-                        stopped: false
-                        identifiedOnAnotherDevice: SWObject(config.identifiedOnAnotherDevice).default(false)
-                    }
-
-    dateTimeObjects = SwrveGetTimestamp()
-    m.installDate = CreateObject("roDateTime")
-    m.joinedDate = CreateObject("roDateTime")
-
-    m.eventsQueue = []
-    m.eventsQAQueue = []
-    'm.swrveNextFlush = CreateObject("roDateTime").AsSeconds() 
-
-    m.userResources = []
-    m.userCampaigns = {}
-
-    m.resourceManager = Invalid
-
-    'used to control campaign refresh
-    m.eventsSentOrQueuedRecently = true
-
-    m.numberOfMessagesShown = 0
-    m.startSessionAsSeconds = CreateObject("roDateTime").AsSeconds()
-    m.swrveNextUpdateCampaignsAndResources = CreateObject("roDateTime").AsSeconds()
-
-    'Development: Benchmarking
-    SwrvePrintLoadingTimeFromTimestamp("Swrve() init dateTimeObjects", dateTimeObjects)
-
-    m.swrve_config.uniqueDeviceID = checkOrWriteQADeviceID()
-
-    SwrveSaveStringToPersistence(SwrveConstants().SWRVE_USER_ID_KEY, m.swrve_config.userID)
-
-    'Development: Benchmarking
-    globalObjects = SwrveGetTimestamp()
-
-    
-    m.global.setField("SwrveDebug", m.swrve_config.debug)
-    m.global.setField("SwrveIsQAUser", m.swrve_config.isQAUser)
-    m.global.observeField("swrveShowIAM", "onShowSwrveIAM")
-    m.global.setField("swrveCurrentIAM", {})
-    m.global.observeField("SwrveShutdown", "SwrveShutdown")
-    m.global.setField("swrveAssetsReady", false)
-    m.global.observeField("SwrveEvent", "onSwrveGlobalEvent")
-    m.global.observeField("SwrveClickEvent", "onSwrveClickEvent")
-    m.global.observeField("SwrvePurchaseEvent", "onSwrvePurchaseEvent")
-    m.global.observeField("SwrveUserUpdate", "onSwrveUserUpdate")
-    m.global.observeField("SwrveImpressionEvent", "SwrveOnImpressionEvent")
-    m.global.setField("swrveSDKHasCustomRenderer", false)
-
-    m.global.observeField("SwrveGetNewResourcesDiff", "SwrveGetResourcesDiff")
-    m.global.observeField("SwrveGlobalCurrencyGiven", "SwrveGlobalCurrencyGiven")
-    m.global.observeField("SwrveGlobalUserUpdateWithDate", "SwrveGlobalUserUpdateWithDate")
-    m.global.observeField("SwrveGlobalIAPWithoutReceipt", "SwrveGlobalIAPWithoutReceipt")
-    m.global.observeField("SwrveGlobalFlushAndClean", "SwrveFlushAndClean")
-
-    'Development: Benchmarking
-    SwrvePrintLoadingTimeFromTimestamp("Swrve() init globalObjects", globalObjects)
-
-    
-    m.global.observeField("SwrveGlobalIdentifyExternalID", "SwrveGlobalIdentify")
-
-    SwrveStartSession()
-
-    'Development: Benchmarking
-    SwrvePrintLoadingTimeFromTimestamp("Swrve() init swrveInit", swrveInit)
+	SWLogInfo("SwrveClient()")
+
+	swrveInit = SwrveGetTimestamp()
+	m.swrve_config = {
+		version: "7"
+		appVersion: SWGetValue(config.appVersion, CreateObject("roAppInfo").GetVersion())
```

Comment:
Depending on how you use a component library you may need to supply your own `appVersion` This will allow developers to do just that. Otherwise it will fallback to the manifest like it did before.